src/python-common/ceph/deployment/service_spec.py

```diff
@@ -1583,3 +1584,49 @@ def copy(self) -> 'TunedProfileSpec':
         # for making deep copies so you can edit the settings in one without affecting the other
         # mostly for testing purposes
         return TunedProfileSpec(self.profile_name, self.placement, self.settings.copy())
+
+
+class CephExporterSpec(ServiceSpec):
+    def __init__(self,
+                 service_type: str = 'ceph-exporter',
+                 sock_dir: Optional[str] = '/',
+                 addrs: str = '',
+                 port: Optional[int] = None,
+                 prio_limit: Optional[int] = 5,
+                 stats_period: Optional[int] = 5,
+                 placement: Optional[PlacementSpec] = None,
+                 unmanaged: bool = False,
+                 preview_only: bool = False,
+                 extra_container_args: Optional[List[str]] = None,
+                 ):
+        assert service_type == 'ceph-exporter'
+
+        super(CephExporterSpec, self).__init__(
+            service_type,
+            placement=placement,
+            unmanaged=unmanaged,
+            preview_only=preview_only,
+            extra_container_args=extra_container_args)
+
+        self.service_type = service_type
+        self.sock_dir = sock_dir or ''
+        self.addrs = addrs
+        self.port = port
+        self.prio_limit = prio_limit
+        self.stats_period = stats_period
+
+    def validate(self) -> None:
+        super(CephExporterSpec, self).validate()
+
+        if not os.path.isdir(self.sock_dir):
+            raise SpecValidationError(
+                    f'Directory does not exist. Got: {self.sock_dir}')
```

Comment:
these checks are run run on a cephadm shell session, I don't think the socket would exist by then! .. the validation here is meant to make sure the spec file is correct (units, values, etc) it shouldn't check any container runtime environment.

Reply:
Oh okay, understood! So is there any other way I can put this validation on `dir`? Here I'll just check type for `prio_limit` & `stats_period`

Reply:
> Oh okay, understood! So is there any other way I can put this validation on `dir`? Here I'll just check type for `prio_limit` & `stats_period`

Here it would be just validation for the spec content itself. If you want to put something on the daemon startup & run then you have to introduce this logic on the cephadm binary side.

Reply:
Ok, thanks! I added the check in cephadm.py